include/alpaka/pltf/PltfUniformCudaHipRt.hpp

```diff
@@ -155,77 +149,48 @@ namespace alpaka
-                        // Return the previous error from cudaStreamCreate.
-                        ALPAKA_CUDA_RT_CHECK(
-                            rc);
-                        // Reset the Error state.
-                        cudaGetLastError();
-                               return false;
-                    }
-                }
 #else
                     hipError_t rc(hipSetDevice(static_cast<int>(iDevice)));
+#endif
 
-                    hipStream_t queue = {};
+                    ALPAKA_API_PREFIX(Stream_t) queue = {};
                     // Create a dummy queue to check if the device is already used by an other process.
-                    // hipSetDevice never returns an error if another process already uses the selected device and gpu compute mode is set "process exclusive".
+                    // cuda/hip-SetDevice never returns an error if another process already uses the selected device and gpu compute mode is set "process exclusive".
                     // \TODO: Check if this workaround is needed!
-                    if(rc == hipSuccess)
+                    if(rc == ALPAKA_API_PREFIX(Success))
                     {
-                        rc = hipStreamCreate(&queue);
+                        rc = ALPAKA_API_PREFIX(StreamCreate)(&queue);
                     }
 
-                    if(rc == hipSuccess)
+                    if(rc == ALPAKA_API_PREFIX(Success))
                     {
                         // Destroy the dummy queue.
-                        ALPAKA_HIP_RT_CHECK(
-                            hipStreamDestroy(
+                        ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK(
+                            ALPAKA_API_PREFIX(StreamDestroy)(
                                 queue));
                         return true;
                     }
                     else
                     {
-                        // Return the previous error from hipStreamCreate.
-                        ALPAKA_HIP_RT_CHECK(
+                        // Return the previous error from cudaStreamCreate.
+                        ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK(
                             rc);
                         // Reset the Error state.
-                        hipGetLastError();
-
-                        return false;
+                        ALPAKA_API_PREFIX(GetLastError)();
+                               return false;
                     }
                 }
-#endif
-
 
 #if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
                 //-----------------------------------------------------------------------------
                 //! Prints all the device properties to std::cout.
                 ALPAKA_FN_HOST static auto printDeviceProperties(
 #if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
-                    cudaDeviceProp const & devProp)
+                    cudaDeviceProp const & devProp
 #else
-                    hipDeviceProp_t const & devProp)
+                    hipDeviceProp_t const & devProp
```

Comment:
So this type is actually called differently, not just a prefix change?

Reply:
yes different type in CUDA and HIP